### PR TITLE
Add Dropdown component

### PR DESCRIPTION
**Pull request by joien1**

### CODE REVIEW

File: packages/hs-react-ui/src/components/Button/Button.tsx [hunk 1]
```diff
@@ -35,7 +33,7 @@ export type ButtonProps = {
   StyledContainer?: string & StyledComponentBase<any, {}>;
   icon?: any;
   isLoading?: any;
-  children?: string | Node;
+  children?: ReactNode;
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/Button/Button.tsx:36]: I've been just putting `any` here since we could potentially let them pass in an array or ReactNodes as well

File: packages/hs-react-ui/src/components/Dropdown/Dropdown.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import { withDesign } from 'storybook-addon-designs';
+import { Dropdown } from './Dropdown';
+
+addDecorator(withDesign);
+const design = {
+    type: 'figma',
+    url: 'https://www.figma.com/file/3r2G00brulOwr9j7F6JF59/Generic-UI-Style?node-id=102%3A28',
+};
+
+const Background = styled.div`background: white; height: 100%;`;
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/Dropdown/Dropdown.stories.tsx:16]: Have you pulled from master? I feel like this isn't necessary anymore.

But either way it stops people from being able to use the storybook background feature to see what the dropdown looks like on a dark background

File: packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+}}
+  
+`;
+const ValueContainer = styled(ButtonContainer)`
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-end;
+  align-items: center;
+  
+  line-height: initial;
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx:30]: does it really break anything to put this change into `Button`?

File: packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+  width: 15rem;
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx:31]: could you add a TODO here to add constants for widths - like `fullWidth: 100%` and `inputWidth: 15rem`?

File: packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+  background: white;
+  position: absolute;
+  max-height: 10rem;
+  overflow-y: scroll;
+  width: 15rem;
+  border: 0.5px solid ${colors.blueCharcoal25};
+`;
+
+const OptionItem = styled.div`
+  height: 2rem;
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx:57]: does this look funky when there's two lines of text? Maybe just padding here would be better

File: packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+  display: flex;
+  align-items: center;
+  
+  font-family: Montserrat,Roboto,sans-serif;
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx:61]: I feel like this should be on the container - and maybe just be `{fonts.body}` to make sure it gets imported when this is the only component brought in

File: packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+
+  clearable?: boolean,
+  elevation?: number,
+  multi?: boolean,
+  name: string,
+  onClear?: () => void,
+  onSelect?: (selected: string | Array<string>) => void,
+  options: Array<string>,
+  tabIndex?: number,
+  values?: Array<string>,
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx:94]: for my own education, what's the difference between this and `string[]`?

File: packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+  });
+
+  const handleBlur = useCallback((e: React.FocusEvent) => {
+    e.preventDefault();
+    const target = e.nativeEvent.relatedTarget as HTMLElement | null;
+    // check if we're focusing on something we don't control
+    if (!target || (target.id && !target.id.startsWith(state.id))) {
+      setState(curState => ({ ...curState, isOpen: false }));
+    }
+  }, [state.id]);
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx:128]: We should probably add an onBlur callback to go with this guy

File: packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx [hunk 9]
```diff
@@ -0,0 +1,279 @@
...
+      >
+        <StyledValueItem>{((values.length && values) || state.selectedValues).join(', ')}</StyledValueItem>
+        <ValueIconContainer>
+          {clearable && (
+            <Icon
+              // TODO Fix issue with Icon not correctly exposing its onClick on its declaration file
+              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+              // @ts-ignore
+              onClick={handleClear}
+              onFocus={(e: FocusEvent) => e.stopPropagation()}
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx:239]: The mdi docs say they don't want icon to have onclick for whatever reason - But you could wrap it in a div with those attached to it. (we probably want the padding around the icon to be clickable as well really)

File: packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx [hunk 10]
```diff
@@ -0,0 +1,279 @@
...
+            >
+              <CheckContainer>
+                {state.selectedValues.includes(opt)
+                  && (
+<Icon
+  path={mdiCheck}
+  size={0.75}
+  color={readableColor(colors.blueCharcoal50, colors.success)}
+/>
+)
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx:268]: this indention lookin real weird on gh

File: packages/hs-react-ui/src/components/Table/Table.tsx [hunk 11]
```diff
@@ -470,8 +470,7 @@ const Table = ({
             >
               {Object.keys(copiedColumns).map(headerColumnKey => {
                 const RenderedCell = copiedColumns[headerColumnKey].cellComponent || StyledCell;
-                const breakPointHit =
-                  width > (copiedColumns[headerColumnKey].minTableWidth || Infinity);
+                const breakPointHit =                  width > (copiedColumns[headerColumnKey].minTableWidth || Infinity);
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/Table/Table.tsx:473]: not sure if prettier is drunk

File: packages/hs-react-ui/src/components/Button/Button.tsx (hunk 1 above)
Comment (joien1) [reply packages/hs-react-ui/src/components/Button/Button.tsx:36]: I believe ReactNode covers everything:
https://github.com/typescript-cheatsheets/react-typescript-cheatsheet/blob/cc2ec67e0b36b7e838df8f2a54b4cd84df6c78cd/README.md#useful-react-prop-type-examples

File: packages/hs-react-ui/src/components/Dropdown/Dropdown.stories.tsx (hunk 2 above)
Comment (joien1) [reply packages/hs-react-ui/src/components/Dropdown/Dropdown.stories.tsx:16]: Nope I forgot to remove this.

File: packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx (hunk 3 above)
Comment (joien1) [reply packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx:30]: Doesn't seem like it. Generally I am trying to avoid modifying components (*goes on to modify Button), but maybe I should not have that mindset right now.

Comment (joien1) [reply packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx:30]: Made the change to button and seems to not affect anything.

File: packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx (hunk 5 above)
Comment (joien1) [reply packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx:57]: It does look funky - made change.

File: packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx (hunk 8 above)
Comment (joien1) [reply packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx:128]: Good point.

File: packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx (hunk 7 above)
Comment (joien1) [reply packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx:94]: https://stackoverflow.com/questions/38239579/what-is-the-difference-arraystring-and-string

TL;DR Nothing, except Array<string> can be extended, string[] can't - that's why I just tend to prefer using Array<string>

File: packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx (hunk 9 above)
Comment (joien1) [reply packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx:239]: Interesting, I wonder why that's the case. I've moved it into a div.

File: packages/hs-react-ui/src/components/Table/Table.tsx (hunk 11 above)
Comment (joien1) [reply packages/hs-react-ui/src/components/Table/Table.tsx:473]: It's drunk. I'll fix it >.<

File: packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx (hunk 6 above)
Comment (joien1) [reply packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx]: Done.

File: packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx (hunk 4 above)
Comment (joien1) [reply packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx:33]: Done.

File: packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx (hunk 8 above)
Comment (joien1) [reply packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx]: Done.

File: packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx (hunk 10 above)
Comment (joien1) [reply packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx]: Prettier had a few too many.

File: packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx (hunk 7 above)
Comment (aVileBroker) [reply packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx]: neat

File: packages/hs-react-ui/src/components/Dropdown/Dropdown.stories.tsx [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+};
+
+storiesOf('Dropdown', module).add(
+  'Basic',
+  () => {
+    const options = ['Apple', 'Banana', 'Grapefruit', 'Grape', 'Orange', 'Watermelon'];
+    return (
+      <>
+        <label htmlFor="fave-fruit">What's your favorite fruit?</label>
+        <Dropdown
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/Dropdown/Dropdown.stories.tsx:22]: one thing that concerns me is that the container should be the source of truth

Comment (joien1) [reply packages/hs-react-ui/src/components/Dropdown/Dropdown.stories.tsx:22]: Can you elaborate on that?

File: packages/hs-react-ui/src/components/Dropdown/Dropdown.stories.tsx [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+          clearable={boolean('clearable', false)}
+          elevation={number('elevation', 2, { range: true, min: 0, max: 5, step: 1 })}
+          multi={boolean('multi', false)}
+          name="fave-fruit"
+          onBlur={action('onBlur')}
+          onClear={action('onClear')}
+          onSelect={action('onSelect')}
+          options={options}
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/Dropdown/Dropdown.stories.tsx:30]: should we include a `values` knob? It'd just be a select with the values - and we wouldn't be able show multiselects properly but idk. May be better to put into the docs

File: packages/hs-react-ui/src/enums/colors.ts [hunk 14]
```diff
@@ -9,6 +9,11 @@ enum colors {
 
   success = '#698F00',
   destructive = '#d00',
+
+  blueCharcoal25 = 'rgba(211, 214, 215, 0.25)',
+  blueCharcoal50 = 'rgba(211, 214, 215, 0.5)',
+  blueCharcoal75 = 'rgba(211, 214, 215, 0.75)',
+  blueCharcoal100 = 'rgba(211, 214, 215, 1)',
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/enums/colors.ts:16]: these are grayDark with different opacities, not the headstorm colors that match these. We also want the users to be able to change these eventually and calling it "blue" and "charcoal" will be weird when they use a warm grey base.

I vote grayDark25, grayDark50, etc

Comment (joien1) [reply packages/hs-react-ui/src/enums/colors.ts:16]: Sure - there's a comment in figma's styles above the color I used that said blueCharcoal, which is why I used that name here. Maybe figma should also be changed.

File: packages/hs-react-ui/src/components/Dropdown/Dropdown.stories.tsx (hunk 13 above)
Comment (joien1) [reply packages/hs-react-ui/src/components/Dropdown/Dropdown.stories.tsx:32]: I've included one (doesn't work for multi-select), but yeah we'll definitely want to put this in the docs anyway.

File: packages/hs-react-ui/src/enums/colors.ts (hunk 14 above)
Comment (aVileBroker) [reply packages/hs-react-ui/src/enums/colors.ts]: Sounds good to me
